src/manager/framework/api/http_api.go

```diff
@@ -372,6 +406,104 @@ func (api *AppService) GetAppVersion(request *restful.Request, response *restful
+	app, err := api.Scheduler.InspectApp(request.PathParameter("app_id"))
+	if err != nil {
+		logrus.Errorf("Get app versions error: %s", err.Error())
+		response.WriteError(http.StatusNotFound, err)
+		return
+	}
+	serviceDiscoveries := getServiceDiscoveris(app)
+	response.WriteEntity(serviceDiscoveries)
+}
+
+func (api *AppService) GetAppServiceDiscoveriesMD5(request *restful.Request, response *restful.Response) {
+	app, err := api.Scheduler.InspectApp(request.PathParameter("app_id"))
+	if err != nil {
+		logrus.Errorf("Get app versions error: %s", err.Error())
+		response.WriteError(http.StatusNotFound, err)
+		return
+	}
+	serviceDiscoveries := getServiceDiscoveris(app)
+	sd, err := json.Marshal(serviceDiscoveries)
+	if err != nil {
+		logrus.Errorf("Fails to Marshal serviceDiscoveries: %s", err.Error())
+		response.WriteError(http.StatusBadRequest, err)
+		return
+	}
+	sdMD5 := md5.Sum(sd)
+	response.WriteEntity(hex.EncodeToString(sdMD5[:]))
+}
+
+func (api *AppService) GetAllServiceDiscoveries(request *restful.Request, response *restful.Response) {
+	appFilterOptions := scheduler.AppFilterOptions{}
+	allServiceDiscoveries := make([]types.ServiceDiscovery, 0)
+	for _, app := range api.Scheduler.ListApps(appFilterOptions) {
+		serviceDiscoveries := getServiceDiscoveris(app)
+		allServiceDiscoveries = append(allServiceDiscoveries, serviceDiscoveries...)
+	}
+	response.WriteEntity(allServiceDiscoveries)
+}
+
+func (api *AppService) GetAllServiceDiscoveriesMD5(request *restful.Request, response *restful.Response) {
+	appFilterOptions := scheduler.AppFilterOptions{}
+	allServiceDiscoveries := make([]types.ServiceDiscovery, 0)
+	for _, app := range api.Scheduler.ListApps(appFilterOptions) {
+		serviceDiscoveries := getServiceDiscoveris(app)
+		allServiceDiscoveries = append(allServiceDiscoveries, serviceDiscoveries...)
+	}
+	sd, err := json.Marshal(allServiceDiscoveries)
+	if err != nil {
+		logrus.Errorf("Fails to Marshal serviceDiscoveries: %s", err.Error())
+		response.WriteError(http.StatusBadRequest, err)
+		return
+	}
+	sdMD5 := md5.Sum(sd)
+	response.WriteEntity(hex.EncodeToString(sdMD5[:]))
+}
+
+func getServiceDiscoveris(app *state.App) []types.ServiceDiscovery {
+	slots := app.GetSlots()
+	serviceDiscoveries := make([]types.ServiceDiscovery, 0)
+	if app.Mode == state.APP_MODE_FIXED {
+		for _, slot := range slots {
+			serviceDiscovery := types.ServiceDiscovery{
+				TaskId:  slot.ID,
+				AppId:   slot.App.ID,
+				AppMode: string(slot.App.Mode),
+				IP:      slot.Ip,
+			}
+			serviceDiscoveries = append(serviceDiscoveries, serviceDiscovery)
+		}
+	} else {
+		for _, slot := range slots {
+			serviceDiscovery := types.ServiceDiscovery{
+				TaskId:  slot.ID,
+				AppId:   slot.App.ID,
+				AppMode: string(slot.App.Mode),
+				IP:      slot.AgentHostName,
+			}
+			serviceDiscovery.Url = strings.ToLower(strings.Replace(slot.ID, "-", ".", -1))
```

Comment:
should add domain here. 

better way would be add a method for slot return domain